xarray/core/variable.py

```diff
@@ -235,7 +240,10 @@ def _as_array_or_item(data):
 
     TODO: remove this (replace with np.asarray) once these issues are fixed
     """
+    if isinstance(data, sparse.COO):
+        data = data.todense()
```

Comment:
If I  understand correctly, this will automatically coerce sparse -> dense arrays with `DataArray.values`. I don't think this is a good idea -- see https://github.com/pydata/xarray/issues/3245

Reply:
Agreed. it was a temp thing to make development easier.